src/syslog_mcp/tools/analysis/timeline_analyzer.py

```
@@ -0,0 +1,661 @@
+    
+    if len(timeline_data) >= 2:
+        # Calculate time span
+        first_timestamp = timeline_data[0]["timestamp"]
+        last_timestamp = timeline_data[-1]["timestamp"]
+        
+        try:
+            first_dt = datetime.fromisoformat(first_timestamp.replace("Z", "+00:00"))
+            last_dt = datetime.fromisoformat(last_timestamp.replace("Z", "+00:00"))
+            duration = last_dt - first_dt
+            features["duration_hours"] = round(duration.total_seconds() / 3600, 1)
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)